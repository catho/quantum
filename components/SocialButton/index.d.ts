import React from 'react';
import Button, { ButtonProps } from '../Button';

interface SocialButtonProps extends ButtonProps {
  provider: string;
  title: string;
}

export default class SocialButton extends React.Component<SocialButtonProps> {}
