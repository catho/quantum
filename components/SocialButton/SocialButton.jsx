import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../Button';
import FacebookIcon from './sub-components/FacebookIcon';
import GoogleIcon from './sub-components/GoogleIcon';
import { theme as defaultTheme } from '../shared';

const BUTTON_SIZE = '40px';
const BUTTON_PADDING = defaultTheme.spacing.xsmall;

const BaseButton = styled(Button)`
  width: ${BUTTON_SIZE};
  height: ${BUTTON_SIZE};
  border-radius: 9px;
  cursor: pointer;
  overflow: hidden;
  outline: none;
`;

const GoogleButton = styled(BaseButton)`
  padding: ${BUTTON_PADDING - 1}px;
`;

const FacebookButton = styled(BaseButton)`
  border: none;
  background-color: #3c5193;
  padding: ${BUTTON_PADDING}px;
`;

const SocialButton = ({ provider, ...props }) => {
  switch (provider) {
    case 'facebook':
      return (
        <FacebookButton {...props}>
          <FacebookIcon title="facebook-button" size="24" />
        </FacebookButton>
      );
    case 'google':
      return (
        <GoogleButton {...props} stroked skin="neutral">
          <GoogleIcon title="google-button" size="24" />
        </GoogleButton>
      );
    default:
      return <Button />;
  }
};

SocialButton.propTypes = {
  provider: PropTypes.oneOf(['facebook', 'google']).isRequired,
  onClick: PropTypes.func,
};

SocialButton.defaultProps = {
  onClick: () => {},
};

export default SocialButton;
