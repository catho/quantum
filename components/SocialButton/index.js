import SocialButton from './SocialButton';

export default SocialButton;
