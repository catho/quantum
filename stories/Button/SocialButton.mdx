import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import { Header } from '../shared';
import SocialButton from '../../components/SocialButton';
import { 
  Google,
  Facebook
} from './SocialButton.stories.jsx';

<Meta title="Button/Social" component={SocialButton}  />

<Header title="Social Button">
    These buttons are used to perform an action related to their
    respective social media. An example is realizing a login using the OAuth 
    API from Google.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import SocialButton from '@catho/quantum/SocialButton';
```

<Canvas>
  <Story id="button-social--google"/>
</Canvas>

## <a id="api"></a>API

<ArgsTable of={SocialButton} />

## <a id="examples"></a>Examples

### Facebook

<Canvas>
  <Story id="button-social--facebook"/>
</Canvas>