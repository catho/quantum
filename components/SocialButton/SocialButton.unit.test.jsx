import React from 'react';
import renderer from 'react-test-renderer';
import SocialButton from './SocialButton';

describe('SocialButton component', () => {
  it('Should match the snapshot', () => {
    expect(
      renderer.create(<SocialButton provider="google" />).toJSON(),
    ).toMatchSnapshot();
  });

  it('when provider prop is set to facebook', () => {
    expect(
      renderer.create(<SocialButton provider="facebook" />).toJSON(),
    ).toMatchSnapshot();
  });
});
