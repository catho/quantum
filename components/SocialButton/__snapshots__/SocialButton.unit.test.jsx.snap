// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton component Should match the snapshot 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  font-size: 16px;
  line-height: 24px;
  min-height: 40px;
  padding: 4px 16px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #ffffff;
  color: #424242;
  border: 2px solid #424242;
  box-shadow: 0px 3px 1px -2px rgba(153,153,153,0.2),0px 2px 2px 0px rgba(153,153,153,0.14),0px 1px 5px 0px rgba(153,153,153,0.12);
}

.c1:hover {
  box-shadow: 0px 2px 4px -1px rgba(25,25,25,0.2),0px 4px 5px 0px rgba(25,25,25,0.14),0px 1px 10px 0px rgba(25,25,25,0.12);
  background-color: #f2f2f2;
  border-color: #191919;
}

.c1:focus,
.c1:focus-within {
  box-shadow: 0px 2px 4px -1px rgba(66,66,66,0.2),0px 4px 5px 0px rgba(66,66,66,0.14),0px 1px 10px 0px rgba(66,66,66,0.12);
}

.c1:active {
  box-shadow: 0px 5px 5px -3px rgba(25,25,25,0.2),0px 8px 10px 1px rgba(25,25,25,0.14),0px 3px 14px 2px rgba(25,25,25,0.12);
  background-color: #f2f2f2;
}

.c1 .c3 function (_ref5) {
  -webkit-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -moz-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -ms-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  return: "\\n          font-size: ".concat(fontSizes[size],";\\n          margin-right: ").concat(xxsmall,"px;\\n        ");
}

.c1 .c3 function (_ref5) var fontSizes = {
  xsmall: "".concat(medium,"px"), small:"".concat(medium,"px"), medium:"".concat(large,"px"), large:"".concat(large,"px"), xlarge:"".concat(large,"px");
}

.c2 {
  display: inline-block;
}

.c0 {
  width: 40px;
  height: 40px;
  border-radius: 9px;
  cursor: pointer;
  overflow: hidden;
  outline: none;
  padding: 7px;
}

<button
  className="c0 c1"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  <svg
    aria-labelledby="google-title"
    className="c2"
    height="24"
    id="Google_Icon"
    role="img"
    version="1.1"
    viewBox="0 0 128 128"
    width="24"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <title
      id="google-title"
    >
      google-button
    </title>
    <g
      id="_x31__stroke"
    >
      <g
        id="Google"
      >
        <rect
          clipRule="evenodd"
          fill="none"
          fillRule="evenodd"
          height={128}
          width={128}
        />
        <path
          clipRule="evenodd"
          d="M27.585,64c0-4.157,0.69-8.143,1.923-11.881L7.938,35.648    C3.734,44.183,1.366,53.801,1.366,64c0,10.191,2.366,19.802,6.563,28.332l21.558-16.503C28.266,72.108,27.585,68.137,27.585,64"
          fill="#FBBC05"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M65.457,26.182c9.031,0,17.188,3.2,23.597,8.436L107.698,16    C96.337,6.109,81.771,0,65.457,0C40.129,0,18.361,14.484,7.938,35.648l21.569,16.471C34.477,37.033,48.644,26.182,65.457,26.182"
          fill="#EA4335"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M65.457,101.818c-16.812,0-30.979-10.851-35.949-25.937    L7.938,92.349C18.361,113.516,40.129,128,65.457,128c15.632,0,30.557-5.551,41.758-15.951L86.741,96.221    C80.964,99.86,73.689,101.818,65.457,101.818"
          fill="#34A853"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M126.634,64c0-3.782-0.583-7.855-1.457-11.636H65.457v24.727    h34.376c-1.719,8.431-6.397,14.912-13.092,19.13l20.474,15.828C118.981,101.129,126.634,84.861,126.634,64"
          fill="#4285F4"
          fillRule="evenodd"
        />
      </g>
    </g>
  </svg>
</button>
`;

exports[`SocialButton component when provider prop is set to facebook 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  font-size: 16px;
  line-height: 24px;
  min-height: 40px;
  padding: 4px 16px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #1250C4;
  color: #ffffff;
  border: 2px solid #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(153,153,153,0.2),0px 2px 2px 0px rgba(153,153,153,0.14),0px 1px 5px 0px rgba(153,153,153,0.12);
}

.c1:hover {
  box-shadow: 0px 2px 4px -1px rgba(0,47,123,0.2),0px 4px 5px 0px rgba(0,47,123,0.14),0px 1px 10px 0px rgba(0,47,123,0.12);
  background-color: #002F7B;
  border-color: #002F7B;
}

.c1:focus,
.c1:focus-within {
  box-shadow: 0px 2px 4px -1px rgba(18,80,196,0.2),0px 4px 5px 0px rgba(18,80,196,0.14),0px 1px 10px 0px rgba(18,80,196,0.12);
}

.c1:active {
  box-shadow: 0px 5px 5px -3px rgba(0,47,123,0.2),0px 8px 10px 1px rgba(0,47,123,0.14),0px 3px 14px 2px rgba(0,47,123,0.12);
  background-color: #002F7B;
}

.c1 .c3 function (_ref5) {
  -webkit-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -moz-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  -ms-var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  var: size = _ref5.size, _ref5$theme$spacing = _ref5.theme.spacing, xxsmall = _ref5$theme$spacing.xxsmall, medium = _ref5$theme$spacing.medium, large = _ref5$theme$spacing.large;
  return: "\\n          font-size: ".concat(fontSizes[size],";\\n          margin-right: ").concat(xxsmall,"px;\\n        ");
}

.c1 .c3 function (_ref5) var fontSizes = {
  xsmall: "".concat(medium,"px"), small:"".concat(medium,"px"), medium:"".concat(large,"px"), large:"".concat(large,"px"), xlarge:"".concat(large,"px");
}

.c2 {
  display: inline-block;
}

.c0 {
  width: 40px;
  height: 40px;
  border-radius: 9px;
  cursor: pointer;
  overflow: hidden;
  outline: none;
  border: none;
  background-color: #3c5193;
  padding: 8px;
}

<button
  className="c0 c1"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  <svg
    aria-labelledby="facebook-title"
    className="c2"
    height="24"
    role="img"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="facebook-title"
    >
      facebook-button
    </title>
    <g>
      <rect
        fill="none"
        height={402}
        id="canvas_background"
        width={582}
        x="-1"
        y="-1"
      />
    </g>
    <g>
      <path
        d="m13,10l3,0l0,3l-3,0l0,7l-3,0l0,-7l-3,0l0,-3l3,0l0,-1.255c0,-1.189 0.374,-2.691 1.118,-3.512c0.744,-0.823 1.673,-1.233 2.786,-1.233l2.096,0l0,3l-2.1,0c-0.498,0 -0.9,0.402 -0.9,0.899l0,2.101z"
        fill="#ffffff"
        id="svg_2"
      />
    </g>
  </svg>
</button>
`;
